sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/ScriptTransformationSuite.scala

```diff
@@ -228,6 +228,26 @@ class ScriptTransformationSuite extends SparkPlanTest with SQLTestUtils with Tes
         'e.cast("string")).collect())
     }
   }
+
+  test("SPARK-30973: ScriptTransformationExec should wait for the termination of script") {
+    (0 until 10).foreach { _ =>
+      assume(TestUtils.testCommandAvailable("/bin/bash"))
+
+      val rowsDf = Seq("a", "b", "c").map(Tuple1.apply).toDF("a")
+      val e = intercept[SparkException] {
+        val plan =
+          new ScriptTransformationExec(
+            input = Seq(rowsDf.col("a").expr),
+            script = "some_non_existent_command",
```

Comment:
Is it possible to sleep several seconds in the script so we can reproduce the issue stably?

Reply:
I tried some times in my Mac, this issue can not be reproduced by sleeping several seconds in the script. It is easy to reproduce this issue by this way https://github.com/apache/spark/pull/27724#issuecomment-607628004. So I add an UT by this code.

Reply:
> I tried some times in my Mac, this issue can not be reproduced by sleeping several seconds in the script. It is easy to reproduce this issue by this way [#27724 (comment)](https://github.com/apache/spark/pull/27724#issuecomment-607628004). So I add an UT by this code.

You should notice that my UT is is without serde, if you want to add UT, I think you'd better add two type UT (with serde / without serde)

Reply:
Both type UT are added,   please review the latest code.